pkg/controllers/node/node_controller.go

```diff
@@ -193,3 +308,155 @@ func kick(c chan<- interface{}) {
 		// pass
 	}
 }
+
+// generateHostendpointName returns the auto hostendpoint's name.
+func (c *NodeController) generateHostendpointName(nodeName string) string {
+	return fmt.Sprintf("%s-auto-hep", nodeName)
+}
+
+// getHostendpointExpectedIPs returns all of the known IPs on the node resource
+// that should set on the auto hostendpoint.
+func (c *NodeController) getHostendpointExpectedIPs(node *api.Node) []string {
+	expectedIPs := []string{}
+	if node.Spec.BGP != nil {
+		// BGP IPv4 and IPv6 addresses are CIDRs.
+		if node.Spec.BGP.IPv4Address != "" {
+			ip, _, _ := net.ParseCIDROrIP(node.Spec.BGP.IPv4Address)
+			expectedIPs = append(expectedIPs, ip.String())
+		}
+		if node.Spec.BGP.IPv6Address != "" {
```

Comment:
Did you check if HEPs support mixed address family? I suspect they do, but can't remember off the top of my head.

Reply:
The docs specify IPv4 or IPv6 for expectedIPs but I'll check